### PR TITLE
feat(AnimesPortal): add presence

### PR DESCRIPTION
**Pull request by AngeloCore**
Adding a new presence for [Animes portal](https://animes-portal.info/)

### CODE REVIEW

File: websites/A/Animes Portal/presence.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    let uid = paths[1];
+    let eid = paths[2];
+
+    if (uid && !eid) {
+      let name = document.querySelector('body > main.animated > div.wrapper > article.rowView > header.rowView-head > h1.heading').innerHTML;
+      assign(presenceData, {
+        details: `Viewing ${name}`
+      });
+    } else if (uid && eid) {
+      let ep = document.querySelector('body > main.animated > div.wrapper > section.holder > h1.heading > b#num').innerHTML;
```
Comment (AngeloCore) [inline websites/A/Animes Portal/presence.ts:57]: I will make one function for this, and call it here and in line 93, but in the next update

**Review by EncryptedDev — CHANGES_REQUESTED** [review]
Resolve lint issue

**Comment by AngeloCore** [comment]
@EncryptedDev done!

**Comment by ImRodry** [comment]
Sure?
image

**Comment by AngeloCore** [comment]
> Sure? image

Check this [discord message](https://discord.com/channels/493130730549805057/607524579874832446/925321952455958549)

**Comment by AngeloCore** [comment]
I forgot to sync the repo, done

**Comment by AngeloCore** [comment]
<details>
   <summary>Proof that the presence is working:</summary>
   
   
   
   
   
   
   
   
</details>

File: websites/A/Animes Portal/dist/metadata.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    "id": "728512329888825396"
+  },
+  "service": "Animes Portal",
+  "description": {
+    "en": "Anime, manga and Asian culture",
+    "bg": "Аниме, манга и азиатска култура"
+  },
+  "altnames": [
+    "Аниме"
+  ],
```
Comment (Bas950) [inline websites/A/Animes Portal/dist/metadata.json:15]:
```suggestion
```

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/A/Animes Portal/dist/metadata.json (hunk 2 above)
Comment (AngeloCore) [reply websites/A/Animes Portal/dist/metadata.json:15]: what??

**Review by Bas950 — DISMISSED** [review]

**Comment by Bas950** [comment]
Upload proof of it working

File: websites/A/Animes Portal/presence.ts [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+  if (
+    avatar === "https://static.animes-portal.info/assets/images/avatar.svg" ||
+    !avatar
+  )
+    avatar = "defaultav";
+
+  return avatar;
+}
+
+function assign(data: PresenceData, newdata: PresenceData) {
```
Comment (EncryptedDev) [inline websites/A/Animes Portal/presence.ts:227]: No, just assign data normally...

File: websites/A/Animes Portal/presence.ts [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+      else return;
+    }
+  }
+
+  if (presenceData.details) {
+    presence.setActivity(presenceData);
+  } else presence.setActivity();
+});
+
+declare interface iframeData {
```
Comment (EncryptedDev) [inline websites/A/Animes Portal/presence.ts:147]:
```suggestion
 interface IFrameData {
```

File: websites/A/Animes Portal/presence.ts [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+      const image = document
+        .querySelector(
+          "body > main.animated > div.wrapper > article.rowView > aside.aside > div.cover-holder > img.abs"
+        )
+        ?.getAttribute("src");
+      assign(presenceData, {
+        details: `Viewing ${name}`,
+        largeImageKey: image ?? "logo",
+        smallImageKey: image ? "logo" : "",
+        smallImageText: image ? name : ""
```
Comment (EncryptedDev) [inline websites/A/Animes Portal/presence.ts:122]: Use an if, don't set things to empty strings

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

**Comment by AngeloCore** [comment]
@Bas950 @EncryptedDev done!

<details>
   <summary>Proof that the presence is working:</summary>


















</details>

**Comment by ImRodry** [comment]
Nope, not done
Also you should change “View manga” to “Read manga”

File: websites/A/Animes Portal/presence.ts [hunk 6]
```diff
@@ -0,0 +1,365 @@
...
+      );
+      const username = body.querySelector("span.username").textContent;
+      const avatar = parseAvatarFromAttr(
+        body.querySelector("a.avatar").attributes.getNamedItem("style")
+          .textContent
+      );
+
+      presenceData.details = `Messaging ${username}`;
+      presenceData.smallImageKey = avatar;
+      presenceData.smallImageText = `Messaging ${username}`;
```
Comment (EncryptedDev) [inline websites/A/Animes Portal/presence.ts:36]: Repeats details, no thanks

File: websites/A/Animes Portal/presence.ts [hunk 7]
```diff
@@ -0,0 +1,365 @@
...
+  366: "BG sub"
+};
+
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+    largeImageKey: "logo"
+  };
+
+  const pathname: string = document.location.pathname;
+  const paths = pathname.split("/");
```
Comment (EncryptedDev) [inline websites/A/Animes Portal/presence.ts:15]:
```suggestion
  const { pathname } = document.location,
   paths = pathname.split("/");
```

File: websites/A/Animes Portal/presence.ts [hunk 8]
```diff
@@ -0,0 +1,365 @@
...
+  if (!paths[0]) paths.shift();
+
+  if (pathname === "/") {
+    presenceData.details = "Home";
+    presenceData.state = "Viewing the home page";
```
Comment (EncryptedDev) [inline websites/A/Animes Portal/presence.ts:20]:
```suggestion
    presenceData.details = "Viewing the home page";
```

File: websites/A/Animes Portal/presence.ts [hunk 9]
```diff
@@ -0,0 +1,365 @@
...
+        .split(",")
+        .map((x: string) => parseInt(x, 10))
+        .forEach((f: number) => {
+          if (filterCodes[f]) filters.push(filterCodes[f]);
+        });
+    }
+
+    if (paths[1] === "search" && paths[2]) {
+      presenceData.details = `Searching for ${paths[2].replaceAll("-", " ")} ${
+        filters.length ? "With " + filters.join(", ") : ""
```
Comment (EncryptedDev) [inline websites/A/Animes Portal/presence.ts:80]: no empty strings please

File: websites/A/Animes Portal/presence.ts [hunk 10]
```diff
@@ -0,0 +1,365 @@
...
+      }`;
+    } else if (paths[1]) {
+      presenceData.details = `Viewing animes that starts with the keyboard ${paths[1].toUpperCase()}`;
+    } else
+      presenceData.details = `Viewing animes ${
+        filters.length ? "With " + filters.join(", ") : ""
```
Comment (EncryptedDev) [inline websites/A/Animes Portal/presence.ts:86]: no empty strings please

**Comment by AngeloCore** [comment]
> Also you should change “View manga” to “Read manga”

It's view manga when you check info about the manga, like:
image

**Comment by AngeloCore** [comment]
> Nope, not done

What to do next?

**Comment by ImRodry** [comment]
Fix all the eslint issues

**Comment by AngeloCore** [comment]
> Fix all the eslint issues

The "warnings"?

File: websites/A/Animes Portal/presence.ts [hunk 11]
```diff
@@ -0,0 +1,365 @@
...
+        body.querySelector("a.avatar").attributes.getNamedItem("style")
+          .textContent
+      );
+
+      presenceData.details = `Messaging ${username}`;
+      presenceData.smallImageKey = avatar;
+      presenceData.smallImageText = `Messaging ${username}`;
+    }
+  } else if (pathname === "/chat") {
+    presenceData.details = "Chatting in the chat";
```
Comment (EncryptedDev) [inline websites/A/Animes Portal/presence.ts:39]: More appropriate string?

File: websites/A/Animes Portal/presence.ts [hunk 12]
```diff
@@ -0,0 +1,365 @@
...
+        "div.content > div.holder > span.heading"
+      ).textContent;
+
+      const avatar = parseAvatarFromAttr(
+        body.querySelector("a.avatar").attributes.getNamedItem("style")
+          .textContent
+      );
+
+      presenceData.details = `Viewing otaku ${username}`;
+      presenceData.largeImageKey = avatar;
```
Comment (EncryptedDev) [inline websites/A/Animes Portal/presence.ts:59]: `avatar` is only used once

File: websites/A/Animes Portal/presence.ts [hunk 13]
```diff
@@ -0,0 +1,365 @@
...
+    } else
+      presenceData.details = `Viewing animes ${
+        filters.length ? "With " + filters.join(", ") : ""
+      }`;
+  } else if (pathname.startsWith("/anime/")) {
+    const uid = paths[1];
+    const eid = paths[2];
+
+    if (uid && !eid) {
+      const name = document.querySelector(
```
Comment (EncryptedDev) [inline websites/A/Animes Portal/presence.ts:93]: Add types
```suggestion
      const name = document.querySelector<HTMLHeadingElement>(
```

File: websites/A/Animes Portal/presence.ts [hunk 14]
```diff
@@ -0,0 +1,365 @@
...
+        "body > main.animated > div.wrapper > article.rowView > header.rowView-head > h1.heading"
+      ).innerHTML;
+      const image = document
+        .querySelector(
```
Comment (EncryptedDev) [inline websites/A/Animes Portal/presence.ts:97]:
```suggestion
        .querySelector<HTMLDivElement>(
```

File: websites/A/Animes Portal/presence.ts [hunk 15]
```diff
@@ -0,0 +1,365 @@
...
+          "body > main.animated > div.wrapper > section.holder > div.player > div.holder > div.vplayer"
+        )
+        ?.classList.contains("playing");
+
+      if (!playing) {
+        presenceData.details = `Watching ${name}`;
+        presenceData.state = `Episode ${episode}`;
+        presenceData.buttons = [
+          {
+            label: `Watch ${name}`,
```
Comment (EncryptedDev) [inline websites/A/Animes Portal/presence.ts:129]: This _could_ surpass the char limit
Watch along? Watch video? etc

File: websites/A/Animes Portal/presence.ts [hunk 16]
```diff
@@ -0,0 +1,365 @@
...
+    const filters: string[] = [];
+
+    if (filter) {
+      filter
+        .split(",")
+        .map((x: string) => parseInt(x, 10))
+        .forEach((f: number) => {
+          if (filterCodes[f]) filters.push(filterCodes[f]);
+        });
+    }
```
Comment (EncryptedDev) [inline websites/A/Animes Portal/presence.ts:150]: Seen this multiple times, why don't you use a function instead of being repetitive?

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/A/Animes Portal/iframe.ts [hunk 17]
```diff
@@ -0,0 +1,10 @@
+const iframe = new iFrame();
+iframe.on("UpdateData", async () => {
+  const video: HTMLVideoElement = document.querySelector("#html5player");
+  if (video && video.duration && !isNaN(video.duration))
```
[the remark this answers is not in the document]
Comment (AngeloCore) [reply websites/A/Animes Portal/iframe.ts:4]: Here i have to check for `video` too, should i use `(video && !isNaN(video?.duration))`?

[the remark this answers is not in the document]
Comment (ImRodry) [reply websites/A/Animes Portal/iframe.ts:4]: Please learn how optional chaining works

**Comment by AngeloCore** [comment]
Done, did i miss something?

File: websites/A/Animes Portal/presence.ts (hunk 11 above)
Comment (EncryptedDev) [reply websites/A/Animes Portal/presence.ts:31]:
```suggestion
    presenceData.details = "Chatting";
```

File: websites/A/Animes Portal/iframe.ts [hunk 18]
```diff
@@ -0,0 +1,11 @@
+const iframe = new iFrame();
+iframe.on("UpdateData", async () => {
+  const video = document.querySelector<HTMLVideoElement>("#html5player");
+  if (video && !isNaN(video.duration)) {
```
Comment (EncryptedDev) [inline websites/A/Animes Portal/iframe.ts:4]: learn how optional chaining works
```suggestion
  if (!isNaN(video?.duration)) {
```

File: websites/A/Animes Portal/presence.ts [hunk 19]
```diff
@@ -0,0 +1,294 @@
...
+        ),
+        username = body.querySelector("span.username").textContent;
+
+      presenceData.smallImageKey = parseAvatarFromAttr(
+        body.querySelector("a.avatar").attributes.getNamedItem("style")
+          .textContent
+      );
+
+      presenceData.details = `Messaging ${username}`;
+      presenceData.smallImageText = username;
```
Comment (EncryptedDev) [inline websites/A/Animes Portal/presence.ts:28]: No need to display the username twice as it is displayed in presenceData.details on the line above
```suggestion
```

File: websites/A/Animes Portal/presence.ts [hunk 20]
```diff
@@ -0,0 +1,294 @@
...
+        ).textContent;
+
+      presenceData.largeImageKey = parseAvatarFromAttr(
+        body.querySelector("a.avatar").attributes.getNamedItem("style")
+          .textContent
+      );
+
+      presenceData.details = `Viewing otaku ${username}`;
+      presenceData.smallImageKey = "logo";
+      presenceData.smallImageText = username;
```
Comment (EncryptedDev) [inline websites/A/Animes Portal/presence.ts:52]: Ditto
```suggestion
```

File: websites/A/Animes Portal/presence.ts [hunk 21]
```diff
@@ -0,0 +1,294 @@
...
+    }
+  } else if (pathname.startsWith("/animes")) {
+    if (paths[1] === "search" && paths[2])
+      presenceData.details = `Searching for ${paths[2].replaceAll("-", " ")}`;
+    else if (paths[1])
+      presenceData.details = `Viewing animes that starts with the keyboard ${paths[1].toUpperCase()}`;
```
Comment (EncryptedDev) [inline websites/A/Animes Portal/presence.ts:58]: This is incredibly long, simplify the string then use details and state

File: websites/A/Animes Portal/presence.ts [hunk 22]
```diff
@@ -0,0 +1,294 @@
...
+          .textContent
+      );
+
+      presenceData.details = `Viewing otaku ${username}`;
+      presenceData.smallImageKey = "logo";
+      presenceData.smallImageText = username;
+    }
+  } else if (pathname.startsWith("/animes")) {
+    if (paths[1] === "search" && paths[2])
+      presenceData.details = `Searching for ${paths[2].replaceAll("-", " ")}`;
```
Comment (EncryptedDev) [inline websites/A/Animes Portal/presence.ts:56]:
```suggestion
      presenceData.details = "Searching for";
      presenceData.state = paths[2].replaceAll("-", " ")
```

File: websites/A/Animes Portal/presence.ts [hunk 23]
```diff
@@ -0,0 +1,294 @@
...
+    else if (paths[1])
+      presenceData.details = `Viewing animes that starts with the keyboard ${paths[1].toUpperCase()}`;
+    else presenceData.details = "Viewing animes";
+  } else if (pathname.startsWith("/anime/")) {
+    const [uid, eid] = paths;
+
+    if (uid && !eid) {
+      const name = document.querySelector<HTMLHeadingElement>(
+          "body > main.animated > div.wrapper > article.rowView > header.rowView-head > h1.heading"
+        ).innerHTML,
```
Comment (EncryptedDev) [inline websites/A/Animes Portal/presence.ts:66]: maybe use `.textContent` instead

File: websites/A/Animes Portal/presence.ts [hunk 24]
```diff
@@ -0,0 +1,294 @@
...
+        image = document
+          .querySelector<HTMLDivElement>(
+            "body > main.animated > div.wrapper > article.rowView > aside.aside > div.cover-holder > img.abs"
+          )
+          ?.getAttribute("src");
+
+      if (image) {
+        presenceData.smallImageKey = "logo";
+        presenceData.smallImageText = "logo";
```
Comment (EncryptedDev) [inline websites/A/Animes Portal/presence.ts:75]: It is a given that the logo is a logo
```suggestion
```

File: websites/A/Animes Portal/presence.ts [hunk 25]
```diff
@@ -0,0 +1,294 @@
...
+      }
+
+      presenceData.details = `Viewing ${name}`;
+      presenceData.largeImageKey = image ?? "logo";
+      presenceData.buttons = [
+        {
+          label: "View",
```
Comment (EncryptedDev) [inline websites/A/Animes Portal/presence.ts:82]: View what?

File: websites/A/Animes Portal/presence.ts [hunk 26]
```diff
@@ -0,0 +1,294 @@
...
+          url: document.location.href
+        }
+      ];
+    } else if (uid && eid) {
+      const { name, episode, part } = getInfo(),
+        playing = document
```
Comment (EncryptedDev) [inline websites/A/Animes Portal/presence.ts:88]: `playing` is only used once

File: websites/A/Animes Portal/presence.ts [hunk 27]
```diff
@@ -0,0 +1,294 @@
...
+            url: document.location.href
+          }
+        ];
+
+        if (part) presenceData.state = presenceData.state += ` (part ${part})`;
+      } else return;
+    }
+  } else if (pathname.startsWith("/movies")) {
+    if (paths[1] === "search" && paths[2])
+      presenceData.details = `Searching for ${paths[2].replaceAll("-", " ")}`;
```
Comment (EncryptedDev) [inline websites/A/Animes Portal/presence.ts:109]:
```suggestion
      presenceData.details = "Searching for";
      presenceData.state = paths[2].replaceAll("-", " ")
```

File: websites/A/Animes Portal/presence.ts [hunk 28]
```diff
@@ -0,0 +1,294 @@
...
+    else if (paths[1])
+      presenceData.details = `Viewing movies that starts with the keyboard ${paths[1].toUpperCase()}`;
```
Comment (EncryptedDev) [inline websites/A/Animes Portal/presence.ts:111]: Same here

File: websites/A/Animes Portal/presence.ts [hunk 29]
```diff
@@ -0,0 +1,294 @@
...
+      ).textContent,
+      image = document
+        .querySelector(
+          "body > main.animated > div.wrapper > article.rowView > aside.aside > div.cover-holder > img.abs"
+        )
+        ?.getAttribute("src");
+
+    if (image) {
+      presenceData.smallImageKey = "logo";
+      presenceData.smallImageText = name;
```
Comment (EncryptedDev) [inline websites/A/Animes Portal/presence.ts:125]: Doesn't make sense?
```suggestion
```

File: websites/A/Animes Portal/presence.ts [hunk 30]
```diff
@@ -0,0 +1,294 @@
...
+    }
+
+    presenceData.details = `Viewing movie ${name}`;
+    presenceData.largeImageKey = image ?? "logo";
+  } else if (pathname.startsWith("/manga")) {
+    if (paths[1] === "search" && paths[2])
+      presenceData.details = `Searching for ${paths[2].replaceAll("-", " ")}`;
```
Comment (EncryptedDev) [inline websites/A/Animes Portal/presence.ts:132]:
```suggestion
      presenceData.details = "Searching for";
      presenceData.state = paths[2].replaceAll("-", " ")
```

File: websites/A/Animes Portal/presence.ts [hunk 31]
```diff
@@ -0,0 +1,294 @@
...
+        const list = document.querySelectorAll(
+            "body > main.animated > ul#path > li > a > span"
+          ),
+          name = list[1].textContent,
+          page = document.querySelector(
+            "body > main.animated > div.wrapper > div.heading > b#num"
+          ).textContent;
+
+        presenceData.details = `Reading manga ${name}`;
+        presenceData.state = `Tom: ${tom}, Chapter: ${between(
```
Comment (EncryptedDev) [inline websites/A/Animes Portal/presence.ts:146]: What's tom?

File: websites/A/Animes Portal/presence.ts [hunk 32]
```diff
@@ -0,0 +1,294 @@
...
+          )
+          ?.getAttribute("src");
+
+      if (name) presenceData.details = `Viewing manga ${name}`;
+
+      presenceData.largeImageKey = image ?? "logo";
+
+      if (image) {
+        presenceData.smallImageKey = "logo";
+        presenceData.smallImageText = name;
```
Comment (EncryptedDev) [inline websites/A/Animes Portal/presence.ts:188]: Same here

File: websites/A/Animes Portal/presence.ts [hunk 33]
```diff
@@ -0,0 +1,294 @@
...
+      }
+
+      presenceData.buttons = [
+        {
+          label: "View manga",
+          url: document.location.href
+        }
+      ];
+    } else if (paths[1])
+      presenceData.details = `Viewing manga that starts with the keyboard ${paths[1].toUpperCase()}`;
```
Comment (EncryptedDev) [inline websites/A/Animes Portal/presence.ts:198]: Ditto

File: websites/A/Animes Portal/presence.ts [hunk 34]
```diff
@@ -0,0 +1,294 @@
...
+    else if (!paths[1]) presenceData.details = "Viewing manga";
+  }
+
+  if (presenceData.details) presence.setActivity(presenceData);
+  else presence.setActivity();
+});
+
+interface iframeData {
```
Comment (EncryptedDev) [inline websites/A/Animes Portal/presence.ts:206]: Already requested this, follow interface naming ocnvention
```suggestion
interface IFrameData {
```

File: websites/A/Animes Portal/presence.ts [hunk 35]
```diff
@@ -0,0 +1,294 @@
...
+  } else {
+    presenceData.smallImageKey = "start";
+    presenceData.smallImageText = "Paused";
+  }
+
+  presenceData.details = `Watching ${epInfo.name}`;
+  presenceData.state = `Episode ${epInfo.episode}`;
+  presenceData.buttons = [
+    {
+      label: `Watch ${name}`,
```
Comment (EncryptedDev) [inline websites/A/Animes Portal/presence.ts:234]: "Watch anime" instead?

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/A/Animes Portal/iframe.ts (hunk 18 above)
Comment (Bas950) [reply websites/A/Animes Portal/iframe.ts:4]: Not sure if that works in isNaN since null is not NaN

Comment (Slowlife01) [reply websites/A/Animes Portal/iframe.ts:4]: It'll be `undefined` not `null`, so it would work

File: websites/A/Animes Portal/presence.ts (hunk 19 above)
Comment (AngeloCore) [reply websites/A/Animes Portal/presence.ts:28]: It's looking better when we have username on `smallImageText`

File: websites/A/Animes Portal/presence.ts (hunk 25 above)
Comment (AngeloCore) [reply websites/A/Animes Portal/presence.ts:82]: image

File: websites/A/Animes Portal/presence.ts (hunk 31 above)
Comment (AngeloCore) [reply websites/A/Animes Portal/presence.ts:146]: its the volume, like the arc
i will rename it to `Volume: ${tom}` or arc, idk

File: websites/A/Animes Portal/presence.ts (hunk 22 above)
Comment (AngeloCore) [reply websites/A/Animes Portal/presence.ts:56]: showing that only in `details` is looking better, already tried

**Comment by AngeloCore** [comment]
> ` presenceData.details = "Chatting";`

See, there is only one global chat in the website, and there is also messages

**Comment by AngeloCore** [comment]
Done, what's next?

File: websites/A/Animes Portal/presence.ts (hunk 19 above)
Comment (EncryptedDev) [reply websites/A/Animes Portal/presence.ts:31]: It is repetitive

File: websites/A/Animes Portal/presence.ts (hunk 25 above)
Comment (EncryptedDev) [reply websites/A/Animes Portal/presence.ts:82]: Use a more appropriate string: view anime? view manga

File: websites/A/Animes Portal/presence.ts (hunk 22 above)
Comment (EncryptedDev) [reply websites/A/Animes Portal/presence.ts:59]: It won't fit on anybody's client lmao

File: websites/A/Animes Portal/presence.ts (hunk 25 above)
Comment (AngeloCore) [reply websites/A/Animes Portal/presence.ts:82]: ...
ok

File: websites/A/Animes Portal/presence.ts (hunk 22 above)
Comment (AngeloCore) [reply websites/A/Animes Portal/presence.ts:59]: ok, lets when is more than 10 characters to be in the `state` and when is less than 10 to be in `details`, ok?

Comment (AngeloCore) [reply websites/A/Animes Portal/presence.ts:59]: image
image

**Comment by AngeloCore** [comment]
@EncryptedDev, am I ready?

File: websites/A/Animes Portal/dist/metadata.json [hunk 36]
```diff
@@ -0,0 +1,42 @@
...
+    "name": "Angelo II",
+    "id": "728512329888825396"
+  },
+  "service": "Animes Portal",
+  "description": {
+    "en": "Anime, manga and Asian culture",
+    "bg": "Аниме, манга и азиатска култура"
+  },
+  "url": "animes-portal.info",
+  "logo": "https://i.imgur.com/I2wctf2.png",
```
Comment (Slowlife01) [inline websites/A/Animes Portal/dist/metadata.json:14]: Must be at least 512x512

**Review by Slowlife01 — DISMISSED** [review]

File: websites/A/Animes Portal/dist/metadata.json (hunk 36 above)
Comment (AngeloCore) [reply websites/A/Animes Portal/dist/metadata.json]: Done

**Comment by AngeloCore** [comment]
@Slowlife01 done!

File: websites/A/Animes Portal/presence.ts [hunk 37]
```diff
@@ -0,0 +1,309 @@
...
+      let str = "Searching for";
+      if (query.length <= 10) str = str += ` ${query}`;
+      else presenceData.state = query;
+
+      presenceData.details = str;
+    } else if (paths[1])
+      presenceData.details = `Viewing animes starting with ${paths[1].toUpperCase()}`;
+    else presenceData.details = "Viewing animes";
+  } else if (pathname.startsWith("/anime/")) {
+    const [_, uid, eid] = paths;
```
Comment (EncryptedDev) [inline websites/A/Animes Portal/presence.ts:68]:
```suggestion
    const [, uid, eid] = paths;
```

File: websites/A/Animes Portal/presence.ts [hunk 38]
```diff
@@ -0,0 +1,309 @@
...
+      ];
+    } else if (paths[1])
+      presenceData.details = `Viewing manga starting with ${paths[1].toUpperCase()}`;
+    else if (!paths[1]) presenceData.details = "Viewing manga";
+  }
+
+  if (presenceData.details) presence.setActivity(presenceData);
+});
+
+interface iFrameData {
```
Comment (EncryptedDev) [inline websites/A/Animes Portal/presence.ts:212]: Third time...
```suggestion
interface IFrameData {
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

**Comment by AngeloCore** [comment]
@EncryptedDev, done senpai!

**Comment by AngeloCore** [comment]
<details>
<summary>Proof that everything is working after the update</summary>




<p>When disable the <code>Show my chats</code> setting</p>

<br>






<p>When is more than 10 chars</P>

<br>






<p>When disable <code>Show thumbnails</code> setting</p>






</details>

File: websites/A/Animes Portal/presence.ts [hunk 39]
```diff
@@ -0,0 +1,309 @@
...
+    {
+      label: "Watch anime",
+      url: document.location.href
+    }
+  ];
+
+  if (epInfo.part)
+    presenceData.state = presenceData.state += ` (part ${epInfo.part})`;
+
+  if (thumb !== "logo" && showThumb === true) {
```
Comment (EncryptedDev) [inline websites/A/Animes Portal/presence.ts:253]:
```suggestion
  if (thumb !== "logo" && showThumb) {
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/A/Animes Portal/dist/metadata.json [hunk 40]
```diff
@@ -0,0 +1,42 @@
...
+    "name": "Angelo II",
+    "id": "728512329888825396"
+  },
+  "service": "Animes Portal",
+  "description": {
+    "en": "Anime, manga and Asian culture",
+    "bg": "Аниме, манга и азиатска култура"
+  },
+  "url": "animes-portal.info",
+  "logo": "https://i.imgur.com/TMTfBFE.png",
```
[the remark this answers is not in the document]
Comment (AngeloCore) [reply websites/A/Animes Portal/dist/metadata.json:14]: image

[the remark this answers is not in the document]
Comment (AngeloCore) [reply websites/A/Animes Portal/dist/metadata.json:14]: Where does it say that it should be 512x512?
image

[the remark this answers is not in the document]
Comment (Slowlife01) [reply websites/A/Animes Portal/dist/metadata.json:14]: image

https://docs.premid.app/en/dev/presence/guidelines#logo

**Comment by AngeloCore** [comment]
done

**Comment by AngeloCore** [comment]
Bread 🍞

File: websites/A/Animes Portal/dist/metadata.json (hunk 40 above)
[the remark this answers is not in the document]
Comment (AngeloCore) [reply websites/A/Animes Portal/dist/metadata.json]: Okay, already changed

File: websites/A/Animes Portal/presence.ts (hunk 11 above)
Comment (Bas950) [reply websites/A/Animes Portal/presence.ts:35]: agreed

File: websites/A/Animes Portal/dist/metadata.json [hunk 41]
```diff
@@ -0,0 +1,42 @@
...
+  "settings": [
+    {
+      "id": "showthumb",
+      "title": "Show thumbnails",
+      "icon": "fas fa-image",
+      "value": true
+    },
+    {
+      "id": "showmessaging",
+      "title": "Show my chats",
```
Comment (EncryptedDev) [inline websites/A/Animes Portal/dist/metadata.json:37]: Use something less personal
```suggestion
      "title": "Show chats",
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/A/Animes Portal/presence.ts (hunk 11 above)
Comment (AngeloCore) [reply websites/A/Animes Portal/presence.ts:35]: See, there is messages in the chat, and one global chat for everyone in the site

**Comment by AngeloCore** [comment]
Um lol
image

**Comment by EncryptedDev** [comment]
> Um lol image

Unrelated to your PR and your changes, ignore it

**Comment by AngeloCore** [comment]
@EncryptedDev could you merge this?

**Review by Bas950 — APPROVED** [review]

**Review by EncryptedDev — APPROVED** [review]

**Comment by AngeloCore** [comment]
Thanks!

**Pull request merged**